bool too_close_side
bool too_close_front
